src/deepsparse/v2/middleware/timer_middleware.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from typing import Any
+
+from deepsparse.v2.middleware.middlewares import MiddlewareCallable
+
+
+class TimerMiddleware(MiddlewareCallable):
+    def __init__(self, call_next: MiddlewareCallable, identifier: str):
+        self.identifier: str = identifier
+        self.call_next: MiddlewareCallable = call_next
+
+    def __call__(self, *args, **kwargs) -> Any:
+        start_time = time.time()
```

Comment:
let's rewrite a little to make it easier to get the fine grained timings we disussed,

```python
class TimerMiddleware(MiddlewareCallable):
    @contextmanager
    def time(self, id: str):
         start time ...
         yield
         end time ...
         self.send({id: time})

    def __call__(self, *args, **kwargs) -> Any:
        with self.time(id=kwargs["inferenence_state"].get("name")):
            result = self.call_next(*args, **kwargs)
        return result
```

then in middeware manager:

```python
class MiddlewareManager:
     @property
     def timer() -> Optional[TimerMiddleware]:
        ...
    @contextmanager
    def time(self, id):
       if self.timer:
            # do timer middleware time()
        else:
            # do nothing
```

that way in an operator we can do (may move middleware manager out of self

```python
class MyOperator():

    def run():
         with self.middleware_manager.time(id="specific_timing"):
              # do something
```

Reply:
let's move timer into a separate PR with this